neo4j-jdbc/src/main/java/org/neo4j/driver/jdbc/CallableStatementImpl.java

```
@@ -48,564 +58,804 @@ final class CallableStatementImpl extends PreparedStatementImpl implements Calla
-		return null;
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public Reader getNCharacterStream(String parameterName) throws SQLException {
-		return null;
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public Reader getCharacterStream(int parameterIndex) throws SQLException {
-		return null;
+		throw newOutParametersNotSupported();
 	}
 
 	@Override
 	public Reader getCharacterStream(String parameterName) throws SQLException {
-		return null;
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setBlob(String parameterName, Blob x) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setClob(String parameterName, Clob x) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setAsciiStream(String parameterName, InputStream x, long length) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setBinaryStream(String parameterName, InputStream x, long length) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setCharacterStream(String parameterName, Reader reader, long length) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setAsciiStream(String parameterName, InputStream x) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setBinaryStream(String parameterName, InputStream x) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setCharacterStream(String parameterName, Reader reader) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
 	}
 
 	@Override
 	public void setNCharacterStream(String parameterName, Reader value) throws SQLException {
-
+		throw new SQLFeatureNotSupportedException();
```

I have been thinking about these (as in the exceptions when we don't support something). I wonder if we should supply a fail-fast mode and a please-try mode. I'm thinking of tools that might "Use" something we don't support. For example, datagrip needed to call a few methods that it did not use other than to check where they were, but it failed so early it did not get to utilise the things it could use. I think being too "Correct" could cause us problems in the future. FWIW, I think that all the methods in this method should fail fast It is more of a comment to think about going forward than this specific PR?

---

Yeah, it is a point to consider. Although, following the specification seems reasonable to me at the moment. We might have to deal with the relevant tools on a case by case basis and deviate when we he have specific reasons to do so. One option could be to have a branch with `fail-fast mode and a please-try mode` modes for such investigation(s).